### ISSUE TITLE
New Idl generation fails when seeds are generated using an external function

### ISSUE THREAD

**Issue by Bhargavamacha** [coral-xyz/anchor#2903]
Consider the following example: 
``` rust
#[account(
    init,
    seeds = [
        CONSTANT_SEED,
        external_function(ix_param).as_slice()
    ],
    bump,
    space = 8 + size,
    payer = payer
)]
```
IDL generation fails saying ix_param not found.

**Comment by acheroncrypto** [comment]
Yeah, it's because the new IDL generation stores PDA seeds in the IDL by default, but there is no way to store `external_function` in the IDL.

**Comment by Bhargavamacha** [comment]
Is there a way to fix it?

**Comment by gwalen** [comment]
There are cases when you are not able to do it inline and need to call a function, like in [#3448](https://github.com/coral-xyz/anchor/issues/3448).
I think there should be a way to disable the resolution of a specific account using some sort of annotation/marker in the instruction definition code.

**Comment by ChewingGlass** [comment]
Sometimes you need to do more complex things in the seeds than the IDL is going to be able to resolve. Seems absurd you have to turn off resolution for the ENTIRE codebase because of one instance of complicated seeds.

**Comment by acheroncrypto** [comment]
> Sometimes you need to do more complex things in the seeds than the IDL is going to be able to resolve. Seems absurd you have to turn off resolution for the ENTIRE codebase because of one instance of complicated seeds.

Yeah, I totally get that and think we should support more cases, but why would a seed need to be so complicated that it cannot be stored in the IDL?

> This issue is basically going to block Helium from migrating to Anchor 30 :/

I'd recommend skipping v0.30 and directly going to v0.31 (https://github.com/coral-xyz/anchor/pull/3259) because it has a lot of improvements over the previous version, including improvements to supported seeds.

**Comment by ChewingGlass** [comment]
A good example here of complex seeds is the way Helium handles calculations on a daily basis. We declare 1 helium epoch = 1 day. We then use that epoch (unix_ts / 60 * 60 * 24) as part of the key in accounts that accumulate data over the course of that day. Figuring out which accumulator you need to work with often requires some nested if statements, just due to the complexity of trying to track metrics over long time horizons. Math for staking usually involves touching 2-3 of these accumulators at different epochs (the epoch the stake ends, the epoch the landrush multiplier ends, and the current epoch)

**Comment by Henry-E** [comment]
Hoping that v0.31.1 fixed this
